public/js/actions/sources.js

```diff
@@ -41,7 +41,8 @@ function newSource(source) {
   return ({ dispatch, getState }) => {
     if (isEnabled("features.sourceMaps") && source.sourceMapURL) {
       const tab = getSelectedTab(getState());
-      loadSourceMapSources(tab, source, dispatch);
+      loadOriginalSources(tab, source)
+        .then(sources => sources.forEach(s => dispatch(newSource(s))));
```

Comment:
it'd be nice to update this to load the source map as an action. Then getting the original sources can be done in the utility

```js
const sourceMap = yield loadSourceMap(source);
const sources = yield loadOriginalSources(sourceMap);
sources.forEach((s) => dispatch(newSource)));
```